src/utils.c

```diff
@@ -42,49 +42,61 @@ is_xdg_current_desktop (const gchar* desktop)
     }
     g_strfreev (desktop_names);
   }
+
+  if (session != NULL)
+  {
+    const gchar *desktop_session = g_getenv ("DESKTOP_SESSION");
+
+    if (desktop_session != NULL && g_str_equal(desktop_session, session))
+    {
+        return TRUE;
+    }
+
+  }
+
```

Comment:
Don't we need unit tests for this? Please add unit tests to this PR that cover the above code section.